### PR TITLE
Hide dashboard actions on public/static embed dashboards

### LINKED ISSUE #34395: Execute an Action on embedded Dashboard fails with 404 not found API call
### Describe the bug

Given there is a dashboard with an action.

- Using the action in the dashboard directly on Metabase works as expected. With the API call pattern:
`https://METABASE_URL/api/dashboard/DASHBOARD_ID/dashcard/ACTION_ID/execute`
- Using the action in the dashboard that is embedded on website does not work with a 404 not found API response.
`https://METABASE_URL/api/dashboard/DASHBOARD_EMBEDDED_TOKEN/dashcard/ACTION_ID/execute`

Seems that the embedded routes have missing action execute on API.

### To Reproduce

- Go to an embedded dashboard with an action.
- Click the action.
- Add parameters
- Execute the action
- The action fails with not found error and network error on webclient console with a 404 response

### Expected behavior

Action could be executed from embedded dashboard

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "fr-FR",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.7+7-LTS",
    "java.vendor": "Azul Systems, Inc.",
    "java.vendor.url": "http://www.azul.com/",
    "java.version": "17.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.7+7-LTS",
    "os.name": "Linux",
    "os.version": "5.4.0-135-generic",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "heroku",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.6 (Debian 14.6-1.pgdg110+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-07-28",
      "tag": "v0.46.6.4",
      "branch": "release-x.46.6.x",
      "hash": "7c60aca"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Blocking usage of actions on embedded dashboard

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/34395

### Description

This PR hides the dashboard actions on public dashboard and static embeds.

Actually we already prevent those actions from running in dashboard in the first place, but this PR hides the actions so there's no confusion whatsoever that we don't allow actions to be executed on public and static embed dashboards.

### How to verify

1. Create a model
2. Turn on the model actions setting on the database of the model in step 1)
3. Create an action of that model, you could use basic actions
4. Create a dashboard and add the action from step 3)
5. Share this dashboard via static embed or public link
6. Test that the action isn't visible in those cases

#### Before
image

#### After
image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/support/helpers/e2e-dashboard-helpers.js [hunk 1]
```diff
@@ -262,7 +262,7 @@ export function dashboardSaveButton() {
 }
 
 /**
- * @param {Object=} option
+ * @param {Object} option
```
Comment (WiNloSt) [inline e2e/support/helpers/e2e-dashboard-helpers.js:265]: I don't know why making `Object` optional with `=` suffix breaks VS Code autocompletion here. I tested that using `[option]` also works.

File: e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js [hunk 2]
```diff
@@ -277,6 +282,113 @@ const MODEL_NAME = "Test Action Model";
...
+
+          cy.log("Assert public dashboard");
+          cy.findByRole("heading", { name: dashboardDetails.name }).should(
+            "be.visible",
+          );
+          /**
+           * cy.button("Create") doesn't work because the label is
+           * `Actions are not enabled for this database` for disabled actions
+           */
+          cy.findByRole("main").findByText("Create").should("not.exist");
```
Comment (WiNloSt) [inline e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js:334]: I was a bit surprised using `cy.button("Create")` doesn't fail the test before introducing the fix. Turned out the action label is different when it's disabled.

File: e2e/support/helpers/e2e-dashboard-helpers.js [hunk 3]
```diff
@@ -359,7 +359,9 @@ export function getLinkCardDetails({
...
+ * @param {number=} option.col
+ * @param {number=} option.row
```
Comment (WiNloSt) [inline e2e/support/helpers/e2e-dashboard-helpers.js:364]: add missing types for other properties.

File: frontend/src/metabase/public/containers/PublicDashboard.jsx [hunk 4]
```diff
@@ -185,6 +189,7 @@ class PublicDashboard extends Component {
             <DashboardContainer>
               <DashboardGridConnected
                 {...this.props}
+                dashboard={assoc(dashboard, "dashcards", visibleDashcards)}
```
Comment (WiNloSt) [inline frontend/src/metabase/public/containers/PublicDashboard.jsx:192]: The fix is to hide all action dashcards.

File: frontend/src/metabase/public/containers/PublicDashboard.jsx [hunk 5]
```diff
@@ -157,6 +158,9 @@ class PublicDashboard extends Component {
       ? getDashboardActions(this, { ...this.props, isPublic: true })
       : [];
 
+    const isNotAction = dashcard => !dashcard.action;
+    const visibleDashcards = (dashboard?.dashcards ?? []).filter(isNotAction);
+
```
Comment (kulyk) [inline frontend/src/metabase/public/containers/PublicDashboard.jsx:163]: Could we please use the [`isActionDashCard` type guard](https://github.com/metabase/metabase/blob/83d79b24cfdb97f42f08302f4b5bca818ae911c4/frontend/src/metabase/dashboard/utils.ts#L97) here?

**Review by kulyk — APPROVED** [review]
LGTM! Please consider using an existing `isActionDashCard` type guard

File: frontend/src/metabase/public/containers/PublicDashboard.jsx (hunk 5 above)
Comment (WiNloSt) [reply frontend/src/metabase/public/containers/PublicDashboard.jsx:163]: Awesome. This is very handy. Thanks ❤️

**Review by albertoperdomo — APPROVED** [review]

**Comment by WiNloSt** [comment]
@metabase-bot backport release-x.49.x

**Pull request merged**